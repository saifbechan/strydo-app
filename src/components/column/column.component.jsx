import React from 'react';

import { Card } from '../card/card.component';
import { MoveCard } from '../move-card/move-card.component';

import { ColumnContainer, H3Container, CardsContainer } from './column.styles';

export const Column = ({ id, title, backgroundColor, cards }) => (
  <ColumnContainer id={id}>
    <H3Container backgroundColor={backgroundColor}>{title}</H3Container>
    <CardsContainer>
      <MoveCard idx={0} columnId={id} />
      {cards.map((card, idx) => (
        <Card key={card.id} id={card.id} idx={idx} columnId={id}>
          {card.content}
        </Card>
      ))}
    </CardsContainer>
  </ColumnContainer>
);

export default Column;
