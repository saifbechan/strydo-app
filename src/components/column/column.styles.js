import styled from 'styled-components';

export const ColumnContainer = styled.div`
  border: 1px solid grey;
`;

export const H3Container = styled.h3`
  background-color: ${({ backgroundColor }) => `${backgroundColor}`};
  font-size: 15px;
  margin: 0;
  padding: 10px;
`;

export const CardsContainer = styled.div`
  border-top: 1px solid grey;
`;
