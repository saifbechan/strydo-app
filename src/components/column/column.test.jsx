import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { useDrag, useDrop } from 'react-dnd';
import 'jest-styled-components';

import { Column } from './column.component';

import mockStore from '../../redux/store.mock';

jest.mock('react-dnd');

useDrag.mockImplementation(() => [{ isDragging: false }, () => {}]);
useDrop.mockImplementation(() => [{ isOver: false }, () => {}]);

it('renders correctly', () => {
  const props = {
    id: 'abc-123',
    title: 'xyz',
    backgroundColor: '#FFF',
    cards: [
      {
        id: '123-abc',
        content: 'content'
      }
    ]
  };

  const { asFragment } = render(
    <Provider store={mockStore()}>
      <Column {...props} />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
