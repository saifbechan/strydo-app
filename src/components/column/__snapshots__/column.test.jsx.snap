// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .c6 {
  cursor: pointer;
  margin-top: 15px;
  padding: 5px;
  text-align: center;
  width: 100%;
}

.c6:hover {
  color: grey;
}

.c3 {
  height: 5px;
  padding: 5px;
}

.c4 {
  background: none;
  border: none;
  height: 100%;
}

.c5 {
  border: 1px solid silver;
  margin: 0 5px;
  padding: 15px 20px;
}

.c5:hover {
  background-color: whitesmoke;
}

.c0 {
  border: 1px solid grey;
}

.c1 {
  background-color: #FFF;
  font-size: 15px;
  margin: 0;
  padding: 10px;
}

.c2 {
  border-top: 1px solid grey;
}

<div
    class="c0"
    id="abc-123"
  >
    <h3
      class="c1"
    >
      xyz
    </h3>
    <div
      class="c2"
    >
      <div
        class="c3"
        data-testid="move-card-handler"
      >
        <div
          class="c4"
        />
      </div>
      <div>
        <div
          class="c5"
          data-testid="card"
          id="123-abc"
        >
          content
          <button
            class="c6"
            data-testid="remove-card-handler"
          >
            - remove card
          </button>
        </div>
        <div
          class="c3"
          data-testid="move-card-handler"
        >
          <div
            class="c4"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
